src/components/views/elements/AppTile.js

```diff
@@ -128,13 +145,18 @@ export default React.createClass({
                 loading: false,
             });
         });
-        window.addEventListener('message', this._onMessage, false);
     },
 
     componentWillUnmount() {
         window.removeEventListener('message', this._onMessage);
     },
 
+    componentDidUpdate(prevProps) {
```

Comment:
This was intentional, so that the _getInitialState function (now renamed to _getNewUrlState) could use the current properties, whether called from the initial load (getInitialState), or when the properties have been updated. By switching it around we have to pass the properties to be used ... which seems clunkier to me, and needs a comment to explain why we are passing props to the function, rather than just using "this.props".

Anyway, I've gone and changed it around, as suggested...